File: xml/jruby/update/xml_update.rb
#! /usr/bin/jruby
# -*- coding: utf-8 -*-
#
#	update/xml_update.rb
#
#					May/18/2011
#
# ------------------------------------------------------------
include	Java
import	java.lang.System
#
require "rexml/document"
include REXML
#
load '/var/www/data_base/common/ruby_common/file_io.rb'
load '/var/www/data_base/common/ruby_common/xml_manipulate.rb'
load '/var/www/data_base/common/ruby_common/text_manipulate.rb'
# ------------------------------------------------------------
puts	"*** 開始 *** xml_update.rb ***"
System.out.println("*** 開始 ***")
#
xml_file = ARGV[0]
#
id_in = ARGV[1]
population_in = ARGV[2].to_i
#
puts id_in,population_in
#
xml_str = read_string_proc(xml_file)

dict_aa=xml_to_dict_proc(xml_str)
dict_update_proc(dict_aa,id_in,population_in)
#
dict_display_proc(dict_aa)
#
str_out=dict_to_xml_proc(dict_aa)
#
file_write_proc(xml_file,str_out)
#
#
puts	"*** 終了 *** xml_update.rb ***"
System.out.println("*** 終了 ***")
#
# ------------------------------------------------------------

File: xml/narwhal/update/xml_update.js
#! /usr/local/bin/narwhal
// ------------------------------------------------------------------
//
//	xml_update.js
//
//						May/09/2011
//
// ------------------------------------------------------------------
var text_manipulate = require ('/var/www/data_base/common/narwhal_common/text_manipulate.js');
var xml_manipulate = require ('/var/www/data_base/common/narwhal_common/xml_manipulate.js');
// ------------------------------------------------------------------
print	("*** 開始 ***");

var args = require('system').args;
var file_xml = args[1];
var id_in = args[2];
var population_in = args[3];
print	(id_in + "\t" + population_in);

var str_in = readFile(file_xml);


var dict_aa =  xml_manipulate.xml_to_dict_proc (str_in);

dict_aa = text_manipulate.dict_update_proc (dict_aa,id_in,population_in);

text_manipulate.dict_display_proc (dict_aa);

var out_str = xml_manipulate.dict_to_xml_proc (dict_aa);

text_manipulate.string_write_proc (file_xml,out_str);

print	("*** 終了 ***");
// ------------------------------------------------------------------
